### PR TITLE
Fix Map representation.

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Review by lostella — DISMISSED** [review]

File: src/gluonts/itertools.py [hunk 1]
```diff
@@ -193,7 +193,7 @@ def __len__(self):
         return len(self.iterable)
 
     def __repr__(self):
-        return f"Map(data={self.iterable!r})"
+        return f"Map(iterable={self.iterable!r})"
```
Comment (lostella) [inline src/gluonts/itertools.py]: How about the `fn` argument though?

Comment (jaheba) [reply src/gluonts/itertools.py]: I guess we could also add that. Then let's just make this a dataclass.

**Comment by lostella** [comment]
Should `__repr__` go then?

**Comment by jaheba** [comment]
Of course!

**Review by lostella — APPROVED** [review]

**Pull request merged**